code/modules/tier_weapon_spawns/tier_spawn.dm

```diff
@@ -0,0 +1,16 @@
+/obj/effect/landmark/weapon_spawn/proc/spawn_weapon(/obj/item/weapon/weapon_to_spawn)
+	new weapon_to_spawn (src)
+
+/obj/effect/landmark/weapon_spawn/proc/spawn_associated_ammo(obj/item/weapon/gun/gun_to_spawn)
+	for(var/i in 1 to 3) //hardcoded 3 mags.
+		new gun_to_spawn.current_mag
+
+//pistols and knives
+/obj/effect/landmark/weapon_spawn/proc/choose_weapon()
+	weapon_to_spawn = pick(weapon_list)
+
+	spawn_weapon(weapon_to_spawn)
```

Comment:
I wouldn't use the proc here just spawn the weapon instead and assign it to var/obj/item/weapon/gun anyway, even though the type could be wrong you can just return early in the type check below instead of assigning it twice.